datahub_sap_hana/ingestion.py

```diff
@@ -1,40 +1,216 @@
+                )
+                continue
+
+            if not self.config.schema_pattern.allowed(lineage.dependent_view):
+                self.report.report_dropped(
+                    f"{lineage.dependent_schema}.{lineage.dependent_view}"
+                )
+                continue
+
+            key = (lineage.dependent_view, lineage.dependent_schema)
+
+            lineage_elements[key].append(
+                mce_builder.make_dataset_urn(
+                    self.platform,
+                    self.config.get_identifier(
+                        lineage.source_schema, lineage.source_table
+                    ),
+                    self.config.env,
+                )
+            )
+
+        return lineage_elements
+
+    def _get_view_lineage_workunits(self) -> Iterable[MetadataWorkUnit]:
+        """Creates MetadataWorkUnit objects for table lineage based on the downstream and downstream objects from the query results.
+        Returns an iterable MetadataWorkUnit that are emitted to Datahub.
+        """
+
+        lineage_elements = self._get_view_lineage_elements()
+
+        if not lineage_elements:
+            logger.debug("No lineage elements returned.")
+            return None
+
+        for key, source_tables in lineage_elements.items():
+            dependent_view, dependent_schema = key
+
+            urn = mce_builder.make_dataset_urn(
+                self.platform,
+                self.config.get_identifier(
+                    dependent_schema,
+                    dependent_view,
+                ),
+                self.config.env,
+            )
+
+            lineage_mce = mce_builder.make_lineage_mce(source_tables, urn)
+            print(lineage_mce)
+            for item in mcps_from_mce(lineage_mce):
+                wu = item.as_workunit()
+                self.report.report_workunit(wu)
+                yield wu
+
+
+if __name__ == "__main__":
+    config = HanaConfig(
+        username="system",
+        password="HXEHana1",
+        host_port="localhost:39017",
+        scheme="hana",
+
+    )
+    source = HanaSource(config, PipelineContext(run_id="test-run"))
+    lineage_elements = source._get_view_lineage_elements()
+    for table_objects, lineage_urn in lineage_elements.items():
+        print(table_objects)
+
+        for key, source_tables in lineage_elements.items():
+            dependent_view, dependent_schema = key
+
+            urn = mce_builder.make_dataset_urn(
+                source.platform,
+                source.config.get_identifier(
+                    dependent_schema,
+                    dependent_view,
+                ),
+                source.config.env,
+            )
+
+            lineage_mce = mce_builder.make_lineage_mce(source_tables, urn)
+            for item in mcps_from_mce(lineage_mce):
+                print(item.aspect)
+                wu = item.as_workunit()
+                source.report.report_workunit(wu)
```

Comment:
do we really need this? can't we just use the console sink with the datahub-cli?

Reply:
i can remove this, was using it for the initial tests